include/osg/ref_ptr

```diff
@@ -52,6 +59,14 @@ class ref_ptr
             return *this;
         }
 
+#if __cplusplus >= 201103L
+        template<class Other> ref_ptr& operator = (ref_ptr<Other>&& rp)
```

Comment:
Move assignment should also be marked `noexcept`

Reply:
Why?

Reply:
https://github.com/isocpp/CppCoreGuidelines/blob/036324/CppCoreGuidelines.md#c66-make-move-operations-noexcept

This CppCon 2019 video is good for learning more 
https://www.youtube.com/watch?v=St0MNEU5b0o

Reply:
This is always a good practice. Specifically for vectors this is a good practice because vector move-assignment is only noexcept if the elements are nothrow-move-assignable:

https://github.com/llvm-mirror/libcxx/blob/78d6a7767ed57b50122a161b91f59f19c9bd0d19/include/vector#L2929-L2932

Reply:
The problem here is that `unref` potentially can throw exceptions. It can add element to a vector for example. I've changed the code to guarantee exception safety for ref_ptr state. If I add noexcept process will terminate when exception is thrown. Whether to allow such behaviour is up to this library maintainer.

Reply:
I see, up to @robertosfield then

Reply:
@elsid is correct, an assignment to a ref_ptr<> that already has reference to an object could unref that object, and that could either directly throw an exception, or for the destruction of the object could throw an exception.

To be able to make the assignment operator noexcept you'd also have to make unref() a noexcept, and then the destructors that it might call.

However, if the ref_ptr<> is null prior to the move assignment then it'd be safe to make it a noexcept.  You can do this for the move copy constructors, but not the assignment.